### ISSUE TITLE
fix(model/v41): fail closed in ForwardMultimodal/ForwardImagePrompt when the config is V4.1

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13022]
﻿

# fix(model/v41): fail closed in ForwardMultimodal/ForwardImagePrompt when the config is V4.1

``routing
lane: model
paths: ["internal/model/multimodal.go", "internal/model/vision_prompt.go", "internal/model/multimodal_test.go"]
expected_steps: 3
repo: fak
public_issue: anthony-chaudhary/fak#12640
``

## Parent context
Parent: #12640 (native DeepSeek V4.1 Flash execution and physical qualification). Discovered while landing the candidate-source fail-closed guard (6ff7735e9) as an independent verification edge case.

## Problem frame
(*Model).Forward gates every V4.1 pass through 41ForwardAdmitted() (forward.go:483) before calling orwardV41. (*Session).prefillV41/stepV41 do the same. But the governed multimodal entry points call orwardHiddenRows **directly** and never consult V4.1 admission:

- `internal/model/multimodal.go:133` -> `return m.forwardHiddenRows(rows), verdict, nil`
- `internal/model/vision_prompt.go:98` -> `return m.ForwardMultimodal(req)`

`forwardHiddenRows` (forward.go:500-528) dispatches on `isGemma4` / `usesMLAMoELayout` / `isMiniMaxSparseAttn`; a V4.1 config matches none, so it falls through to the generic `m.layer()` path. A V4.1 model therefore emits logits through the generic Q/K/V attention stack — bypassing every V4.1 fail-closed guard (Engram, compressor/indexer, shared-KV source, candidate source, mHC multiplicity) — instead of refusing.

## Why now
The V4.1 native track just hardened admission for all guarded stages; this is the one remaining entry point that can still run a V4.1 config without that gate. Same fail-closed invariant, one seam.

## Core through-line
Make ForwardMultimodal (and therefore ForwardImagePrompt) refuse a V4.1 config with the same typed fail-closed error the scalar/Session paths produce, rather than silently falling through to the generic attention path.

## Gold-plating boundary
No V4.1 multimodal execution, no vision wiring, no changes to the generic (non-V4.1) multimodal path. Refuse only; do not implement.

## Witness
``powershell
go -C ..\fak test ./internal/model -run 'TestV41ForwardMultimodalDeclaredFailsClosed' -count=1
``

### File:Line Seam
`internal/model/multimodal.go:133` (orwardHiddenRows call in ForwardMultimodal)

### Quarantined Fallback Mechanism
A V4.1 ForwardMultimodal request returns an error wrapping ErrV41ForwardStage (or ErrV41NativeUnsupported to match the weightless-fence contract); no silent generic-attention fallback, no mock.

## Scoped Acceptance Criteria
- [ ] [SW-VERIFIED] ForwardMultimodal on a V4.1 config returns a typed fail-closed error, never logits.
- [ ] [SW-VERIFIED] ForwardImagePrompt inherits the refusal.
- [ ] [SW-VERIFIED] A non-V4.1 config's multimodal path is unchanged (no regression).

## Done condition
The scoped acceptance criteria pass with the cited witness; no sibling lane regresses.

## Acceptance gate
Focused witness passes, package tests green, vet and 5-gate boundary checks clean.

## Scope honesty
internal/model/vision_prompt.go:12 documents ForwardMultimodal/NewVisionEncoder have zero production callers, so this is a latent invariant gap rather than a live serving defect. It is still a violable fail-closed invariant and belongs closed before any caller is added.

## Closure binding
The resolving commit cites this issue with Fixes/Ref and carries the repository-qualified lane trailer.

**Comment by anthony-chaudhary** [comment]
Receipt: landed 39fedd373 on \origin/main\ (DCO-signed, \Fixes #13022\, \Ref #12640\). Witness (SW-VERIFIED): \go -C ..\fak test ./internal/model -run TestV41ForwardMultimodalDeclaredFailsClosed -count=1\ => ok; full \./internal/model/...\ green; \go vet ./internal/model\ clean; 5-gate \ak-boundary check --all --fak-dir <worktree>\ => PASS 0 violations. Guard at \internal/model/multimodal.go:133\; reproduction confirmed pre-fix (generic \orward.go:521\ path, panic on missing generic-arch tensor). Non-V4.1 path unchanged.
